ocaml/libs/tracing/tracing.ml

```diff
@@ -832,6 +832,21 @@ module Export = struct
           done
         )
         ()
+
+    let exporter_created = ref None
+
+    let lock = Mutex.create ()
+
+    let main () =
+      Xapi_stdext_threads.Threadext.Mutex.execute lock (fun () ->
+          match !exporter_created with
+          | None ->
+              let tid = create_exporter () in
```

Comment:
We are creating an exporter - but assign to a `tid` - which is a thread ID? Naming could be more congruent such that the function creating a value is related to the variable holding that value. We have here `exporter_created`, `create_exporter` and `tid` which refer to related values but you can't tell this from their names.

Reply:
tid is the thread id of the exporter, it is a temporary variable holding the value of that id. It's just easier to have this temporary variable rather than storing the value in exporter directly, which can be a bit awkward since it is a option value